### PR TITLE
Minor cleanup of gcb stage command

### PR DESCRIPTION
**Pull request by SgtCoDFish**
This is mostly to clean up [this line](https://github.com/cert-manager/release/blob/4d52f6f9027b2330897cc2014e1af8628bc63d22/cmd/cmrel/cmd/gcb_stage.go#L165) which I found very hard to parse (and which has an extra append left over from previous changes).

This will also make it easier to add flags to specify a specific OS or arch to build later, by manually specifying a value for osTargets

### CODE REVIEW

**Review by irbekrm — APPROVED** [review]
This does look like it makes the whole setup easier to read 👍🏼 👍🏼 

/lgtm

File: pkg/release/platforms.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// AllOSes returns a slice of all known operating systems which cert-manager targets
+// including both server and client targets.
+func AllOSes() []string {
+	// initialise set with server platforms
+	platforms := sets.NewString(mapKeys(ServerPlatforms)...)
+
+	// add client platforms
+	platforms = platforms.Insert(mapKeys(ClientPlatforms)...)
+
+	return platforms.List()
```
[the remark this answers is not in the document]
Comment (SgtCoDFish) [reply pkg/release/platforms.go:57]: I tend to agree, yeah; I think "this could be _much_ simpler" applies to most of `cmrel` to be honest :grin: 

Thanks for the review :+1:
